Configurations docs refresh (#4300)
dapr/docs: daprdocs/content/en/operations/configuration/configuration-overview.md

---
type: docs
title: "Dapr configuration"
linkTitle: "Overview"
weight: 100
description: "Overview of Dapr configuration"
---

> With Dapr configuration, you can set options for the sidecar or control plane. 

Reviewer: I suggest that we take this sentence from the conceptual overview to include here
"Dapr configurations are settings and policies that enable you to change both the behavior of individual Dapr applications, or the global behavior of the Dapr control plane system services."
And then add a link here at the start to say "for more information on configuration read https://docs.dapr.io/concepts/configuration-concept/"

Reviewer: Separately can you review the https://docs.dapr.io/concepts/configuration-concept/ docs for working. You may have to change the link on this page to the updated overview doc below anyway.

> ## Sidecar configuration

Reviewer: ```suggestion
## Application configuration
```

Reviewer: Generally we misnamed this by calling it sidecar configuration and not application configuration,  as correctly described in the concepts https://docs.dapr.io/concepts/configuration-concept/ . We should refer to application and control plane configurations. 

I tried to update these and see if there are others that should be addressed.

> ### Set up sidecar configuration

Reviewer: ```suggestion
### Set up application configuration
```

> You can set up sidecar configuration either in self-hosted or Kubernetes mode.

Reviewer: ```suggestion
You can set up application configuration either in self-hosted or Kubernetes mode.
```

{{< tabs "Self-hosted" Kubernetes >}}

 <!-- Self hosted -->
{{% codetab %}}

In self hosted mode, the Dapr configuration is a [configuration file]({{< ref configuration-schema.md >}}) - for example, `config.yaml`. By default, the Dapr sidecar looks in the default Dapr folder for the runtime configuration:
- Linux/MacOs: `$HOME/.dapr/config.yaml`
- Windows: `%USERPROFILE%\.dapr\config.yaml`

> A Dapr sidecar can also apply a configuration by using a `--config` flag to the file path with `dapr run` CLI command.

Reviewer: ```suggestion
An application can also apply a configuration by using a `--config` flag to the file path with `dapr run` CLI command.
```

---

Dapr configurations are settings and policies that enable you to change both the behavior of individual Dapr applications, or the global behavior of the Dapr control plane system services. 

[for more information, read the configuration concept.]({{< ref configuration-concept.md >}})

## Application configuration

### Set up application configuration

You can set up application configuration either in self-hosted or Kubernetes mode.

{{< tabs "Self-hosted" Kubernetes >}}

 <!-- Self hosted -->
{{% codetab %}}

In self hosted mode, the Dapr configuration is a [configuration file]({{< ref configuration-schema.md >}}) - for example, `config.yaml`. By default, the Dapr sidecar looks in the default Dapr folder for the runtime configuration:
- Linux/MacOs: `$HOME/.dapr/config.yaml`
- Windows: `%USERPROFILE%\.dapr\config.yaml`

An application can also apply a configuration by using a `--config` flag to the file path with `dapr run` CLI command.

{{% /codetab %}}

 <!-- Kubernetes -->
{{% codetab %}}

In Kubernetes mode, the Dapr configuration is a Configuration resource, that is applied to the cluster. For example:

```bash
kubectl apply -f myappconfig.yaml
```

> You can use the Dapr CLI to list the Configuration resources.

Reviewer: ```suggestion
You can use the Dapr CLI to list the Configuration resources for applications.
```

    dapr.io/config: "myappconfig"
```

> **Note:** [See all Kubernetes annotations]({{< ref "arguments-annotations-overview.md" >}}) available to configure the Dapr sidecar on activation by sidecar Injector system service.

{{% /codetab %}}

{{< /tabs >}}

> ### Sidecar configuration settings

Reviewer: ```suggestion
### Application configuration settings
```

The following menu includes all of the configuration settings you can set on the sidecar. 

> - [Tracing](#tracing)

Reviewer: Question. Do you think that each of these sub-headings should be also seen in the topic menu on the right. Like this one https://docs.dapr.io/operations/configuration/configuration-overview/#tracing so that you can see the whole list of what is available?

Author: that might need to be an update in the docsy theme... I can investigate and see if it's possible in a separate PR?

Reviewer: Ok

    excludeVerbs: false
```

> In the examples above, the path filter `/orders/{orderID}/items/{itemID}` would return _a single metric count_ matching all the `orderID`s and all the `itemID`s, rather than multiple metrics for each `itemID`. For more information, see [HTTP metrics path matching]({{< ref "metrics-overview.md#http-metrics-path-matching" >}})

Reviewer: The metrics link here is broken. It should point to For more information see [HTTP metrics path matching](https://docs.dapr.io/operations/observability/metrics/metrics-overview/#http-metrics-path-matching). Please try this out
